### PR TITLE
Feature: improve login expired message and clean up storage system

### PR DESCRIPTION
**Pull request by JasonTheAdams**
## Description

The primary goal of this is to provide a more meaningful message to users in the form builder when they got to save and their session has expired. We can later explore a UI to allow the user to log in and not lose their place, but for now we're starting with a good message.

While working on this, I also realized we have some old code for a storage system that no longer works — due to deep WP integration — and cleaned up the storage system a bit.

## Affects

The system for saving, loading, and previewing within the form builder.

## Testing Instructions

Two things to test. First, make sure that the form builder still works, especially saving and previewing forms.

Second, do the following to simulate an expired login session:
1. Open the form builder
2. Open the JS console
3. Press cmd+shift+P to open up a command console and type in "Clear site data", and run the command
4. Make some change in the form
5. Click "update" to save the form
6. An alert should pop up with the new message

## Pre-review Checklist

-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205405424043724

### CODE REVIEW

**Comment by kjohnson** [comment]
The `form-id` query string on the preview URL is being omitted, which is breaking previews.

File: src/FormBuilder/Routes/RegisterFormBuilderPageRoute.php [hunk 1]
```diff
@@ -117,6 +105,12 @@ public function renderPage()
             true
         );
 
+        wp_add_inline_script(
+            '@givewp/form-builder/script',
+            'window.giveStorageData = ' . wp_json_encode($formBuilderViewModel->storageData($donationFormId)) . ';',
```
Comment (kjohnson) [inline src/FormBuilder/Routes/RegisterFormBuilderPageRoute.php:110]: This change is breaking the preview URL - something is different about how the data is localized.

Comment (kjohnson) [reply src/FormBuilder/Routes/RegisterFormBuilderPageRoute.php:110]: `wp_localize_script` passes each value through `html_entity_decode( (string) $value, ENT_QUOTES, 'UTF-8' )` before attaching the data to the window.

Comment (kjohnson) [reply src/FormBuilder/Routes/RegisterFormBuilderPageRoute.php:110]: This seems to be a conflict with calling `esc_url` up the line.

**Review by kjohnson — CHANGES_REQUESTED** [review]
This is breaking the form preview feature due to a change in how the `storageData` is being localized.

File: src/FormBuilder/Routes/RegisterFormBuilderPageRoute.php (hunk 1 above)
Comment (JasonTheAdams) [reply src/FormBuilder/Routes/RegisterFormBuilderPageRoute.php]: Resolved in [3f7cff2](https://github.com/impress-org/givewp/pull/6911/commits/3f7cff25a8bc156a87e0f28923acdb431ea7bf4e)

**Review by kjohnson — APPROVED** [review]
This is good to go!

**Review by rickalday — APPROVED** [review]
Passed manual QA tests.

**Pull request merged**